docs/main_concepts.md

```diff
@@ -81,6 +57,32 @@ add_slider = st.sidebar.slider(
 )
 ```
 
+## Caching
+
+The Streamlit cache allows your app to stay performant even when loading data from the web, manipulating large datasets, or performing expensive computations.
+
+To use the cache, just wrap functions in the [`@st.cache`](api.html#streamlit.cache) decorator:
+
+```python
+@st.cache  # <-- This function will be cached
+def my_slow_function(arg1, arg2):
+    # Do something really slow in here!
+    return the_output
+```
+
+When you mark a function with the [`@st.cache`](api.html#streamlit.cache) decorator, it tells Streamlit that whenever the function is called it needs to check a few things:
+
+1. The input parameters that you called the function with
+2. The value of any external variable used in the function
+3. The body of the function
+4. The body of any function used inside the cached function
+
+If this is the first time Streamlit has seen these items with these exact values and in this exact combination and order, it runs the function and stores the result in a local cache. Think of the cache as a simple in-memory key-value store, where the key is a hash of all of the above and the value is the actual output object passed by reference.
```

Comment:
Update these as mentioned in my first comment, above.

Reply:
Made a best effort based on what I thought needed to go in here.